client/Lykke.Service.Assets.Client/Cache/IDictionaryCache.cs

```diff
@@ -1,18 +1,39 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Threading.Tasks;
+using Common.Log;
 
 namespace Lykke.Service.Assets.Client.Cache
 {
+    /// <summary>
+    /// Simple in-memory client side cache.
+    /// </summary>
     public interface IDictionaryCache<T>
         where T : ICacheItem
     {
+        /// <summary>
+        /// Starts an automatic updater that keeps the cache updated on a background thread.
+        /// </summary>
+        IDisposable StartAutoUpdate(string componentName, ILog log, Func<Task<IEnumerable<T>>> getAllAsync);
```

Comment:
Why someone outside of the cache has to care about the cache freshnes? I think it is a cache responsibility and it must do it itself.

Reply:
the old implementation was an expiring cache so if would check if the cache was expired and I extended it with a cache that keeps up to date in the background. But I agree that this isn't really SOLID. I will have a look at an improved design, but that might also mean some breaking changes.